strings.py

```diff
@@ -1,47 +1,65 @@
-"""With this string:
-'monty pythons flying circus'
-Create a function that returns a sorted string with no duplicate characters
-(keep any whitespace):
-Example: ' cfghilmnoprstuy'
-Create a function that returns the words in reverse order:
-Example: ['circus', 'flying', 'pythons', 'monty']
-Create a function that returns a list of 4 character strings:
-Example: ['mont', 'y py', 'thon', 's fl', 'ying', ' cir', 'cus']
-
-"""
-import pytest
-
-def no_duplicates(a_string):
-    pass
-
-
-def reversed_words(a_string):
-    pass
-
-
-def four_char_strings(a_string):
-    pass
-
-
-def test_no_duplicates():
-    s = 'monty pythons flying circus'
-    assert no_duplicates(s) == ' cfghilmnoprstuy'
-
-
-def test_reversed_words():
-    s = 'monty pythons flying circus'
-    assert reversed_words(s) == ['circus', 'flying', 'pythons', 'monty']
-
-
-def test_four_char_strings():
-    s = 'monty pythons flying circus'
-    assert four_char_strings(s) == ['mont', 'y py', 'thon', 's fl', 'ying', ' cir', 'cus']
-
-
-def main():
-    return pytest.main(__file__)
-
-
-if __name__ == '__main__':
-    main()
-    
+#!/usr/bin/env python3
+"""With this string:
+'monty pythons flying circus'
+Create a function that returns a sorted string with no duplicate characters
+(keep any whitespace):
+Example: ' cfghilmnoprstuy'
+Create a function that returns the words in reverse order:
+Example: ['circus', 'flying', 'pythons', 'monty']
+Create a function that returns a list of 4 character strings:
+Example: ['mont', 'y py', 'thon', 's fl', 'ying', ' cir', 'cus']
+
+"""
+# import pytest
+
+from collections import OrderedDict
+def no_duplicates(a_string):
+    res= "".join(set(a_string));
+    return ''.join(sorted(res));
+print("***sorted and no duplicates:***")
+print(no_duplicates("monty pythons flying circus"));
+print("\n")
+

```

Comment:
# set(a_string) will remove duplicates
    # sorted(sequence) will create a sorted list of sequence
    # ''.join(sequence) will create a single string out of a sequence of strings
    # This can all be done in one line
    return ''.join(sorted(set(a_string)))